### ISSUE TITLE
MarchingCubes::compute_intersection_points is broken

### ISSUE THREAD

**Issue by avaer** [weshoke/efficient-marching-cubes#1]
Thanks for this!

I think the code [here](https://github.com/weshoke/efficient-marching-cubes/blob/master/src/MarchingCubes.cpp#L122) is busted though. It's using local `i` `j` `k` but I think it should be using the instance vars `_i` `_j` `_k`.

The instance vars are [what's actually used in the setting](https://github.com/weshoke/efficient-marching-cubes/blob/master/src/MarchingCubes.cpp#L145). Also those instance vars won't actually be set at all during construction so this will crash depending on your environment.

I think this would be fixed if we use `_i` and friends.

**Comment by weshoke** [comment]
You're right.  Presumably [this](https://github.com/weshoke/efficient-marching-cubes/blob/master/src/MarchingCubes.cpp#L129) should be changed as well.

**Comment by shwilks** [comment]
Thanks also for this code!  I've corrected the error discussed here, I notice there's no licence in this repo but it would be useful if it's ok with you to incorporate this code into an R package?  I would want to make sure that's ok and that I get the code attribution right though, I found it in the first place because this same code is included in scikit for python: https://github.com/scikit-image/scikit-image/blob/master/skimage/measure/mc_meta/MarchingCubes.cpp
